pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/H3Utils.java

```diff
@@ -35,4 +51,50 @@ private H3Utils() {
       throw new RuntimeException("Failed to instantiate H3 instance", e);
     }
   }
+
+  private static Set<Long> coverLineInH3(LineString lineString, int resolution) {
+    Set<Long> coveringH3Cells = new HashSet<>();
+    List<Long> endpointH3Cells = new ArrayList<>();
+    for (Coordinate endpoint : lineString.getCoordinates()) {
+      endpointH3Cells.add(H3_CORE.geoToH3(endpoint.y, endpoint.x, resolution));
+    }
+    for (int i = 0; i < endpointH3Cells.size() - 1; i++) {
+      try {
+        coveringH3Cells.addAll(H3_CORE.h3Line(endpointH3Cells.get(i), endpointH3Cells.get(i + 1)));
+      } catch (LineUndefinedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return coveringH3Cells;
+  }
```

Comment:
Boxing here is unfortunate, use `LongSet` instead.

Reply:
got it. thanks. updated.

Reply:
Thanks